src/promptflow-devkit/promptflow/batch/_batch_engine.py

```diff
@@ -320,39 +340,30 @@ def _copy_previous_run_result(
                     previous_node_run_outputs = {
                         node_info.node: node_info.output for node_info in previous_node_run_infos
                     }
-
                     # Extract aggregation inputs for flow with aggregation node
                     aggregation_inputs = extract_aggregation_inputs(self._flow, previous_node_run_outputs)
 
-                    # Deepcopy to avoid modifying the original object when serializing image
-                    previous_run_output = deepcopy(previous_run_info.output)
-                    previous_run_output_in_line_result = self._multimedia_processor.persist_multimedia_data(
-                        previous_run_output, output_dir
-                    )
-
-                    # Persist previous run info and node run info
-                    self._storage.persist_flow_run(previous_run_info)
+                    # Persist node run info to storage
                     for node_run_info in previous_node_run_infos:
                         self._storage.persist_node_run(node_run_info)
 
-                    # Create LineResult object for previous line result
+                    # Create LineResult object with aggregation inputs and node_run_infos
                     previous_line_result = LineResult(
                         output=previous_run_output_in_line_result,
                         aggregation_inputs=aggregation_inputs,
                         run_info=previous_run_info,
                         node_run_infos=previous_node_run_infos_dict,
                     )
-                    previous_run_results.append(previous_line_result)
-
-            return previous_run_results
+                previous_run_results.append(previous_line_result)
         except Exception as e:
             bulk_logger.error(f"Error occurred while copying previous run result. Exception: {str(e)}")
-            resume_copy_error = ResumeCopyError(
+            raise ResumeCopyError(
                 target=ErrorTarget.BATCH,
                 message_format="Failed to copy results when resuming the run. Error: {error_type_and_message}.",
                 error_type_and_message=f"({e.__class__.__name__}) {e}",
-            )
-            raise resume_copy_error from e
+            ) from e
+
+        return previous_run_results
```

Comment:
Put this return in the end of "try" block since `previous_run_results` is defined in try.